### PR TITLE
Add tool to create RSA keys

### PR DESCRIPTION
**Pull request by stefanberger**
Add tool to create RSA keys

### CODE REVIEW

File: kmip/objects.go [hunk 1]
```diff
@@ -359,33 +461,151 @@ func NewRSAPrivateKey(id, server *string, tlsConfig *tls.Config) (*RSAPrivateKey
...
 	}
 
-	blob, err := k.GetKeyMaterial(KFT_PKCS1, KCT_None, nil, KWT_None)
+	key, err := k.Private()
 	if err != nil {
 		return nil, err
 	}
 
-	key, err := parsePrivateKey(blob, nil)
+	k.public = key.Public().(*rsa.PublicKey)
```
Comment (lumjjb) [inline kmip/objects.go]: Hmm, what's the additional casting here for?

File: tests/clitools.sh [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+sha1plain=$(cat ${textfile} | sha1sum | awk '{print $1}')
+for hash in sha1 sha256 sha384 sha512; do
+	binhash=$(openssl dgst -${hash} ${textfile} | cut -d "=" -f2 | tr -d " " | sed -n 's/[0-9a-f]\{2\}/\\x\0/gp')
+	echo -en "${binhash}" > ${textfile}.${hash}.bin
+done
+
+signaturefile=${workdir}/signature.bin
+pubkeyfile=${workdir}/pubkey.pem
+encryptfile=${workdir}/encryped.bin
+decryptfile=${workdir}/decrytped
```
Comment (lumjjb) [inline tests/clitools.sh]: typos in `encryptfile` and `decryptfile`

File: kmip/objects.go (hunk 1 above)
Comment (stefanberger) [reply kmip/objects.go]: `key.Public()` returns a `crypto.PublicKey` which can only be an RSA key in our case

Comment (lumjjb) [reply kmip/objects.go]: ahh..... i mixed up `rsa.PrivateKey` and `RSAPrivateKey`...

**Pull request merged**